airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -66,8 +66,8 @@ def path(self, **kwargs) -> str:
 
     def next_page_token(self, response: requests.Response) -> str:
         response_data = response.json()
-        if len(response_data["records"]) == self.limit and self.primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
-            return f"WHERE {self.primary_key} > '{response_data['records'][-1][self.primary_key]}' "
+        if len(response_data["records"]) == self.page_size and self.primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
+            return f"WHERE {self.primary_key} >= '{response_data['records'][-1][self.primary_key]}' "
```

Comment:
why aer we comparing PK and not cursor field?

Reply:
Answered in the comment above